Subject: [patch] Add more device IDs for supporting ATI SB700 SATA controller completely
From: Henry Su <henry.su@amd.com>
Patch-Mainline: 2.6.23
References: 291744

openSUSE10.3/10.2 etc may NOT be installed on ATI SB700 when set "onchip SATA
type" in BIOS into "AHCI" mode. There are four SATA related device IDs in
SB700:

0x4390, SATA in IDE mode
0x4391, SATA in AHCI mode
0x4392, SATA in RAID mode with Promise non-Raid5 driver
0x4393, SATA in RAID mode with Promise Raid5 driver

This is different from SB600, there is only one SATA Device ID in SB600, which
is 0x4380.

Signed-off-by: Brandon Philips <bphilips@novell.com>
---
 drivers/ata/ahci.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletion(-)

Index: linux-2.6.22/drivers/ata/ahci.c
===================================================================
--- linux-2.6.22.orig/drivers/ata/ahci.c
+++ linux-2.6.22/drivers/ata/ahci.c
@@ -399,7 +399,10 @@ static const struct pci_device_id ahci_p
 
 	/* ATI */
 	{ PCI_VDEVICE(ATI, 0x4380), board_ahci_sb600 }, /* ATI SB600 */
-	{ PCI_VDEVICE(ATI, 0x4390), board_ahci_sb600 }, /* ATI SB700 */
+	{ PCI_VDEVICE(ATI, 0x4390), board_ahci_sb600 }, /* ATI SB700 IDE */
+	{ PCI_VDEVICE(ATI, 0x4391), board_ahci_sb600 }, /* ATI SB700 AHCI */
+	{ PCI_VDEVICE(ATI, 0x4392), board_ahci_sb600 }, /* ATI SB700 nor-raid5 */
+	{ PCI_VDEVICE(ATI, 0x4393), board_ahci_sb600 }, /* ATI SB700 raid5 */
 
 	/* VIA */
 	{ PCI_VDEVICE(VIA, 0x3349), board_ahci_vt8251 }, /* VIA VT8251 */
